pilot/pkg/model/push_context.go

```diff
@@ -911,7 +911,8 @@ func (ps *PushContext) ServiceAttachedToGateway(hostname string, proxy *Proxy) b
 			}
 		}
 	}
-	return ps.extraServicesForProxy(proxy).Contains(hostname)
```

Comment:
Did the namespace not considered before?

Reply:
yes, if you referenced a namespaced hostname it just didn't work at all.